src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1051,8 +1052,36 @@ def _validate_admin_password(password, os_type):
         raise CLIError(error_msg)
 
 
-def validate_ssh_key(namespace):
-    if namespace.ssh_key_value:
+def validate_ssh_key(namespace, cmd=None):
+    from azure.core.exceptions import HttpResponseError
+    if hasattr(namespace, 'ssh_key_name') and namespace.ssh_key_name:
+        client = _compute_client_factory(cmd.cli_ctx)
+        # --ssh-key-name
+        if not namespace.ssh_key_value and not namespace.generate_ssh_keys:
+            # Use existing key, key must exist
+            try:
+                ssh_key_resource = client.ssh_public_keys.get(namespace.resource_group_name, namespace.ssh_key_name)
+            except HttpResponseError:
+                raise ValidationError('SSH key {} does not exist!'.format(namespace.ssh_key_name))
+            namespace.ssh_key_value = [ssh_key_resource.public_key]
+            logger.info('Get a key from --ssh-key-name successfully')
+        elif namespace.ssh_key_value:
+            parameters = {}
+            parameters['location'] = namespace.location
+            parameters['public_key'] = namespace.ssh_key_value[0]
+            client.ssh_public_keys.create(resource_group_name=namespace.resource_group_name,
+                                          ssh_public_key_name=namespace.ssh_key_name,
+                                          parameters=parameters)
+        elif namespace.generate_ssh_keys:
```

Comment:
when `namespace.generate_ssh_keys` and `namespace.ssh_key_value` both exist, we should throw an error not fall to use  `ssh_key_value`, right?

Reply:
No. I follow the existing logic. Use one, ignore another.

Reply:
which one will be used?